cmd/kubelet/app/server.go

```diff
@@ -500,6 +500,8 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, featureGate f
 	if err != nil {
 		klog.Errorf("unable to register KubeletConfiguration with configz, error: %v", err)
 	}
+	// set klog of kubelet
+	s.Logs.Apply()
```

Comment:
Applying the config to global settings should probably happen super-early in the bootstrap. 
We can start writing logs almost immediately:
https://github.com/kubernetes/kubernetes/blob/0891f69f5ec27cdbcb6336ce3cfe6efdcc1d1e05/cmd/kubelet/app/server.go#L152-L164

Also, are any additional changes required wrt the InitLogs call in https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/kubelet.go#L38?

Reply:
Based on [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#non-goals) the logging format flag just set the format of logging and I think it changing log messages handled by [new klog methods](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#new-klog-methods).  @serathius please correct me if I am wrong.

Reply:
Michael are you proposing then that we move this `--logging-format` flag init/setup/apply to `InitLogs`? This would mean that we would need to make this flag global. 
In https://github.com/kubernetes/kubernetes/pull/89683 we have implemented it as standard flag initialized with component. Like you mentioned that it has problem that we invoke klog before we initialize components. 

Pros of global flag:
* Consistent logging configuration (as initialization happens before component bootstrap)

Cons: 
* All logging flags will need to be kept as global flag 
* LoggerConfig cannot be added to Components (no dynamic config, no control for component owner about this flag) 

I think that having consistent log format is important so we should go with global flag. This would mean that all PRs adding `LoggingConfig` are no longer needed :(.  WDYT?

Reply:
Not necessarily into InitLogs - just that logging config should probably be set before any log messages are written?